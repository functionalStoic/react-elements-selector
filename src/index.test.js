import React from 'react'
import {render} from 'react-testing-library'

import ElementsSelector from './'

describe('ElementsSelector', () => {
  it('renders', () => {
    expect(<ElementsSelector />).toMatchSnapshot()
  })

  it('renders with array', () => {
    const items = [1, 2, 3, 4]
    expect(
      <ElementsSelector
        items={items}
        notSelectable={items.filter(item => !item.selectable)}
      >
        {() => {
          ;<div />
        }}
      </ElementsSelector>
    ).toMatchSnapshot()
  })
})
