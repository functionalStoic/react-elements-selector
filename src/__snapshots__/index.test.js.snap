// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementsSelector renders 1`] = `
<ElementsSelector
  items={Array []}
  notSelectable={Array []}
/>
`;

exports[`ElementsSelector renders with array 1`] = `
<ElementsSelector
  items={
    Array [
      1,
      2,
      3,
      4,
    ]
  }
  notSelectable={
    Array [
      1,
      2,
      3,
      4,
    ]
  }
>
  [Function]
</ElementsSelector>
`;
